src/lib/es5.d.ts

```diff
@@ -1273,7 +1273,20 @@ declare type PropertyDecorator = (target: Object, propertyKey: string | symbol)
 declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
 declare type ParameterDecorator = (target: Object, propertyKey: string | symbol, parameterIndex: number) => void;
 
-declare type PromiseConstructorLike = new <T>(executor: (resolve: (value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void) => PromiseLike<T>;
+/** An object type that can be definitely be awaited. Do not inherit from this type. */
+declare type Awaitable<T> = { then(onfulfilled: (value: T) => any): any; };
+
+/** An object type that may not be awaitable. Do not inherit from this type. */

```

Comment:
If this type isn't supposed to be extended/used, why not just inline the object type inside `Awaited`, below?

Reply:
I've considered that, but this is more readable when getting Quick Info on `Awaited<T>`. I don't even need `Awaitable<T>` if I inline it in the 2-3 places its used but that again reduces readability.

Reply:
I have removed `NonAwaitable`.

Reply:
I decided to also remove `Awaitable<T>` and inline the definition into `Awaited<T>`. The two other places where it was referenced were fine as `PromiseLike<T>`.